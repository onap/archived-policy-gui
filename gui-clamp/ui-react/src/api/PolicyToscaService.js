/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019, 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class PolicyToscaService {
  static getToscaPolicyModels() {
    return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', { method: 'GET', credentials: 'same-origin' })
      .then(function (response) {
        console.debug("getToscaPolicyModels response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getToscaPolicyModels query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getToscaPolicyModels error received", error);
        return {};
      });
  }

  static getToscaPolicyModelYaml(policyModelType, policyModelVersion) {
        return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/yaml/' + policyModelType + "/" + policyModelVersion, {
            method: 'GET',
            credentials: 'same-origin'
        })
            .then(function (response) {
                console.debug("getToscaPolicyModelYaml response received: ", response.status);
                if (response.ok) {
                    return response.json();
                } else {
                    console.error("getToscaPolicyModelYaml query failed");
                    return "";
                }
            })
            .catch(function (error) {
                console.error("getToscaPolicyModelYaml error received", error);
                return "";
            });
 }

 static getToscaPolicyModel(policyModelType, policyModelVersion) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + "/" + policyModelVersion, {
             method: 'GET',
             credentials: 'same-origin'
         })
             .then(function (response) {
                 console.debug("getToscaPolicyModel response received: ", response.status);
                 if (response.ok) {
                     return response.json();
                 } else {
                     console.error("getToscaPolicyModel query failed");
                     return {};
                 }
             })
             .catch(function (error) {
                 console.error("getToscaPolicyModel error received", error);
                 return {};
             });
  }

  static createPolicyModelFromToscaModel(jsonData) {
       return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', {
           method: 'POST',
           credentials: 'same-origin',
           headers: {
             "Content-Type": "a",
           },
           body: JSON.stringify(jsonData)
         })
         .then(function(response) {
           console.debug("createPolicyModelFromToscaModel response received: ", response.status);
           if (response.ok) {
             var message = {
               status: response.status,
               message: 'Tosca Policy Model successfully uploaded'
             };
             return message;
           } else {
             console.error("createPolicyModelFromToscaModel failed");
             return response.text();
           }
         })
         .catch(function(error) {
           console.error("createPolicyModelFromToscaModel error received", error);
           return "";
         });
     }

     static updatePolicyModelTosca(policyModelType, policyModelVersion, jsonData) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + '/' + policyModelVersion, {
             method: 'PUT',
             credentials: 'same-origin',
             headers: {
               "Content-Type": "a",
             },
             body: JSON.stringify(jsonData)
           })
           .then(function(response) {
             console.debug("updatePolicyModelTosca response received: ", response.status);
             if (response.ok) {
               var message = {
                 status: response.status,
                 message: 'Tosca Policy Model successfully uploaded'
               };
               return message;
             } else {
               console.error("updatePolicyModelTosca failed");
               return response.text();
             }
           })
           .catch(function(error) {
             console.error("updatePolicyModelTosca error received", error);
             return "";
           });
       }
}
