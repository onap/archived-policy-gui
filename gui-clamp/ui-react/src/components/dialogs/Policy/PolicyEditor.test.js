/*-
 * ============LICENSE_START=======================================================
 * ONAP POLICY-CLAMP
 * ================================================================================
 * Copyright (C) 2021 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import PolicyEditor from './PolicyEditor';
import { shallow, mount } from 'enzyme';
import PolicyToscaService from '../../../api/PolicyToscaService';

describe('Verify PolicyEditor', () => {
  const fs = require('fs');

  let toscaJson = fs.readFileSync('src/components/dialogs/Policy/toscaData.test.json', { encoding: 'utf8', flag: 'r' })

  const policyProperties = {
    "tca.policy": {
      "domain": "measurementsForVfScaling",
      "metricsPerEventName": [
        {
          "policyScope": "DCAE",
          "thresholds": [
            {
              "version": "1.0.2",
              "severity": "MAJOR",
              "thresholdValue": 200,
              "closedLoopEventStatus": "ONSET",
              "closedLoopControlName": "LOOP_test",
              "direction": "LESS_OR_EQUAL",
              "fieldPath": "$.event.measurementsForVfScalingFields.vNicPerformanceArray[*].receivedTotalPacketsDelta"
            }
          ],
          "eventName": "vLoadBalancer",
          "policyVersion": "v0.0.1",
          "controlLoopSchemaType": "VM",
          "policyName": "DCAE.Config_tca-hi-lo"
        }
      ]
    }
  };


  it('Test the render method', async () => {
    PolicyToscaService.getToscaPolicyModel = jest.fn().mockImplementation(() => {
      return Promise.resolve(toscaJson);
    });

    const component = mount(<PolicyEditor policyModelType="onap.policies.monitoring.tcagen2" policyModelTypeVersion="1.0.0"
                                          policyName="org.onap.new" policyVersion="1.0.0" policyProperties={ policyProperties }
                                          policiesTableUpdateFunction={ () => {
                                          } }/>);
    await PolicyToscaService.getToscaPolicyModel();
    expect(component).toMatchSnapshot();
  });
});
