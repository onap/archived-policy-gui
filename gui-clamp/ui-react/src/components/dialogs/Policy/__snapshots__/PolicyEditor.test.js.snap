// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify PolicyEditor Test the render method 1`] = `
<PolicyEditor
  policiesTableUpdateFunction={[Function]}
  policyModelType="onap.policies.monitoring.tcagen2"
  policyModelTypeVersion="1.0.0"
  policyName="org.onap.new"
  policyProperties={
    Object {
      "tca.policy": Object {
        "domain": "measurementsForVfScaling",
        "metricsPerEventName": Array [
          Object {
            "acmSchemaType": "VM",
            "eventName": "vLoadBalancer",
            "policyName": "DCAE.Config_tca-hi-lo",
            "policyScope": "DCAE",
            "policyVersion": "v0.0.1",
            "thresholds": Array [
              Object {
                "closedLoopControlName": "LOOP_test",
                "closedLoopEventStatus": "ONSET",
                "direction": "LESS_OR_EQUAL",
                "fieldPath": "$.event.measurementsForVfScalingFields.vNicPerformanceArray[*].receivedTotalPacketsDelta",
                "severity": "MAJOR",
                "thresholdValue": 200,
                "version": "1.0.2",
              },
            ],
          },
        ],
      },
    }
  }
  policyVersion="1.0.0"
>
  <styled.div>
    <div
      className="sc-dlnjwi ksxnfR"
    >
      <Alert
        closeLabel="Close alert"
        dismissible={true}
        onClose={[Function]}
        show={false}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "render": [Function],
          }
        }
        variant="success"
      >
        <Fade
          appear={false}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={true}
        >
          <Transition
            addEndListener={[Function]}
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={300}
            unmountOnExit={true}
          />
        </Fade>
      </Alert>
      <Alert
        closeLabel="Close alert"
        dismissible={true}
        onClose={[Function]}
        show={false}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "render": [Function],
          }
        }
        variant="danger"
      >
        <Fade
          appear={false}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={true}
        >
          <Transition
            addEndListener={[Function]}
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={300}
            unmountOnExit={true}
          />
        </Fade>
      </Alert>
      <WithStyles(ForwardRef(TextField))
        defaultValue="org.onap.new"
        id="policyName"
        label="Required"
        onChange={[Function]}
        required={true}
        size="small"
        variant="outlined"
      >
        <ForwardRef(TextField)
          classes={
            Object {
              "root": "MuiTextField-root",
            }
          }
          defaultValue="org.onap.new"
          id="policyName"
          label="Required"
          onChange={[Function]}
          required={true}
          size="small"
          variant="outlined"
        >
          <WithStyles(ForwardRef(FormControl))
            className="MuiTextField-root"
            color="primary"
            disabled={false}
            error={false}
            fullWidth={false}
            required={true}
            size="small"
            variant="outlined"
          >
            <ForwardRef(FormControl)
              className="MuiTextField-root"
              classes={
                Object {
                  "fullWidth": "MuiFormControl-fullWidth",
                  "marginDense": "MuiFormControl-marginDense",
                  "marginNormal": "MuiFormControl-marginNormal",
                  "root": "MuiFormControl-root",
                }
              }
              color="primary"
              disabled={false}
              error={false}
              fullWidth={false}
              required={true}
              size="small"
              variant="outlined"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <WithStyles(ForwardRef(InputLabel))
                  htmlFor="policyName"
                  id="policyName-label"
                >
                  <ForwardRef(InputLabel)
                    classes={
                      Object {
                        "animated": "MuiInputLabel-animated",
                        "asterisk": "MuiInputLabel-asterisk",
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "filled": "MuiInputLabel-filled",
                        "focused": "Mui-focused",
                        "formControl": "MuiInputLabel-formControl",
                        "marginDense": "MuiInputLabel-marginDense",
                        "outlined": "MuiInputLabel-outlined",
                        "required": "Mui-required",
                        "root": "MuiInputLabel-root",
                        "shrink": "MuiInputLabel-shrink",
                      }
                    }
                    htmlFor="policyName"
                    id="policyName-label"
                  >
                    <WithStyles(ForwardRef(FormLabel))
                      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                      classes={
                        Object {
                          "asterisk": "MuiInputLabel-asterisk",
                          "disabled": "Mui-disabled",
                          "error": "Mui-error",
                          "focused": "Mui-focused",
                          "required": "Mui-required",
                        }
                      }
                      data-shrink={true}
                      htmlFor="policyName"
                      id="policyName-label"
                    >
                      <ForwardRef(FormLabel)
                        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                        classes={
                          Object {
                            "asterisk": "MuiFormLabel-asterisk MuiInputLabel-asterisk",
                            "colorSecondary": "MuiFormLabel-colorSecondary",
                            "disabled": "Mui-disabled Mui-disabled",
                            "error": "Mui-error Mui-error",
                            "filled": "MuiFormLabel-filled",
                            "focused": "Mui-focused Mui-focused",
                            "required": "Mui-required Mui-required",
                            "root": "MuiFormLabel-root",
                          }
                        }
                        data-shrink={true}
                        htmlFor="policyName"
                        id="policyName-label"
                      >
                        <label
                          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                          data-shrink={true}
                          htmlFor="policyName"
                          id="policyName-label"
                        >
                          Required
                          <span
                            aria-hidden={true}
                            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                      </ForwardRef(FormLabel)>
                    </WithStyles(ForwardRef(FormLabel))>
                  </ForwardRef(InputLabel)>
                </WithStyles(ForwardRef(InputLabel))>
                <WithStyles(ForwardRef(OutlinedInput))
                  autoFocus={false}
                  defaultValue="org.onap.new"
                  fullWidth={false}
                  id="policyName"
                  label={
                    <React.Fragment>
                      Required
                       *
                    </React.Fragment>
                  }
                  multiline={false}
                  onChange={[Function]}
                >
                  <ForwardRef(OutlinedInput)
                    autoFocus={false}
                    classes={
                      Object {
                        "adornedEnd": "MuiOutlinedInput-adornedEnd",
                        "adornedStart": "MuiOutlinedInput-adornedStart",
                        "colorSecondary": "MuiOutlinedInput-colorSecondary",
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "focused": "Mui-focused",
                        "input": "MuiOutlinedInput-input",
                        "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                        "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                        "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                        "inputMultiline": "MuiOutlinedInput-inputMultiline",
                        "marginDense": "MuiOutlinedInput-marginDense",
                        "multiline": "MuiOutlinedInput-multiline",
                        "notchedOutline": "MuiOutlinedInput-notchedOutline",
                        "root": "MuiOutlinedInput-root",
                      }
                    }
                    defaultValue="org.onap.new"
                    fullWidth={false}
                    id="policyName"
                    label={
                      <React.Fragment>
                        Required
                         *
                      </React.Fragment>
                    }
                    multiline={false}
                    onChange={[Function]}
                  >
                    <WithStyles(ForwardRef(InputBase))
                      autoFocus={false}
                      classes={
                        Object {
                          "adornedEnd": "MuiOutlinedInput-adornedEnd",
                          "adornedStart": "MuiOutlinedInput-adornedStart",
                          "colorSecondary": "MuiOutlinedInput-colorSecondary",
                          "disabled": "Mui-disabled",
                          "error": "Mui-error",
                          "focused": "Mui-focused",
                          "input": "MuiOutlinedInput-input",
                          "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                          "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                          "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                          "inputMultiline": "MuiOutlinedInput-inputMultiline",
                          "marginDense": "MuiOutlinedInput-marginDense",
                          "multiline": "MuiOutlinedInput-multiline",
                          "notchedOutline": null,
                          "root": "MuiOutlinedInput-root",
                        }
                      }
                      defaultValue="org.onap.new"
                      fullWidth={false}
                      id="policyName"
                      inputComponent="input"
                      multiline={false}
                      onChange={[Function]}
                      renderSuffix={[Function]}
                      type="text"
                    >
                      <ForwardRef(InputBase)
                        autoFocus={false}
                        classes={
                          Object {
                            "adornedEnd": "MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd",
                            "adornedStart": "MuiInputBase-adornedStart MuiOutlinedInput-adornedStart",
                            "colorSecondary": "MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary",
                            "disabled": "Mui-disabled Mui-disabled",
                            "error": "Mui-error Mui-error",
                            "focused": "Mui-focused Mui-focused",
                            "formControl": "MuiInputBase-formControl",
                            "fullWidth": "MuiInputBase-fullWidth",
                            "input": "MuiInputBase-input MuiOutlinedInput-input",
                            "inputAdornedEnd": "MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd",
                            "inputAdornedStart": "MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart",
                            "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                            "inputMarginDense": "MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense",
                            "inputMultiline": "MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline",
                            "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                            "marginDense": "MuiInputBase-marginDense MuiOutlinedInput-marginDense",
                            "multiline": "MuiInputBase-multiline MuiOutlinedInput-multiline",
                            "root": "MuiInputBase-root MuiOutlinedInput-root",
                          }
                        }
                        defaultValue="org.onap.new"
                        fullWidth={false}
                        id="policyName"
                        inputComponent="input"
                        multiline={false}
                        onChange={[Function]}
                        renderSuffix={[Function]}
                        type="text"
                      >
                        <div
                          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            autoFocus={false}
                            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                            defaultValue="org.onap.new"
                            disabled={false}
                            id="policyName"
                            onAnimationStart={[Function]}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={true}
                            type="text"
                          />
                          <WithStyles(ForwardRef(NotchedOutline))
                            className="MuiOutlinedInput-notchedOutline"
                            label={
                              <React.Fragment>
                                Required
                                 *
                              </React.Fragment>
                            }
                            labelWidth={0}
                            notched={true}
                          >
                            <ForwardRef(NotchedOutline)
                              className="MuiOutlinedInput-notchedOutline"
                              classes={
                                Object {
                                  "legend": "PrivateNotchedOutline-legend-2",
                                  "legendLabelled": "PrivateNotchedOutline-legendLabelled-3",
                                  "legendNotched": "PrivateNotchedOutline-legendNotched-4",
                                  "root": "PrivateNotchedOutline-root-1",
                                }
                              }
                              label={
                                <React.Fragment>
                                  Required
                                   *
                                </React.Fragment>
                              }
                              labelWidth={0}
                              notched={true}
                            >
                              <fieldset
                                aria-hidden={true}
                                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                              >
                                <legend
                                  className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                                >
                                  <span>
                                    Required
                                     *
                                  </span>
                                </legend>
                              </fieldset>
                            </ForwardRef(NotchedOutline)>
                          </WithStyles(ForwardRef(NotchedOutline))>
                        </div>
                      </ForwardRef(InputBase)>
                    </WithStyles(ForwardRef(InputBase))>
                  </ForwardRef(OutlinedInput)>
                </WithStyles(ForwardRef(OutlinedInput))>
              </div>
            </ForwardRef(FormControl)>
          </WithStyles(ForwardRef(FormControl))>
        </ForwardRef(TextField)>
      </WithStyles(ForwardRef(TextField))>
      <WithStyles(ForwardRef(TextField))
        defaultValue="1.0.0"
        id="policyVersion"
        label="Required"
        onChange={[Function]}
        required={true}
        size="small"
        variant="outlined"
      >
        <ForwardRef(TextField)
          classes={
            Object {
              "root": "MuiTextField-root",
            }
          }
          defaultValue="1.0.0"
          id="policyVersion"
          label="Required"
          onChange={[Function]}
          required={true}
          size="small"
          variant="outlined"
        >
          <WithStyles(ForwardRef(FormControl))
            className="MuiTextField-root"
            color="primary"
            disabled={false}
            error={false}
            fullWidth={false}
            required={true}
            size="small"
            variant="outlined"
          >
            <ForwardRef(FormControl)
              className="MuiTextField-root"
              classes={
                Object {
                  "fullWidth": "MuiFormControl-fullWidth",
                  "marginDense": "MuiFormControl-marginDense",
                  "marginNormal": "MuiFormControl-marginNormal",
                  "root": "MuiFormControl-root",
                }
              }
              color="primary"
              disabled={false}
              error={false}
              fullWidth={false}
              required={true}
              size="small"
              variant="outlined"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <WithStyles(ForwardRef(InputLabel))
                  htmlFor="policyVersion"
                  id="policyVersion-label"
                >
                  <ForwardRef(InputLabel)
                    classes={
                      Object {
                        "animated": "MuiInputLabel-animated",
                        "asterisk": "MuiInputLabel-asterisk",
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "filled": "MuiInputLabel-filled",
                        "focused": "Mui-focused",
                        "formControl": "MuiInputLabel-formControl",
                        "marginDense": "MuiInputLabel-marginDense",
                        "outlined": "MuiInputLabel-outlined",
                        "required": "Mui-required",
                        "root": "MuiInputLabel-root",
                        "shrink": "MuiInputLabel-shrink",
                      }
                    }
                    htmlFor="policyVersion"
                    id="policyVersion-label"
                  >
                    <WithStyles(ForwardRef(FormLabel))
                      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                      classes={
                        Object {
                          "asterisk": "MuiInputLabel-asterisk",
                          "disabled": "Mui-disabled",
                          "error": "Mui-error",
                          "focused": "Mui-focused",
                          "required": "Mui-required",
                        }
                      }
                      data-shrink={true}
                      htmlFor="policyVersion"
                      id="policyVersion-label"
                    >
                      <ForwardRef(FormLabel)
                        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                        classes={
                          Object {
                            "asterisk": "MuiFormLabel-asterisk MuiInputLabel-asterisk",
                            "colorSecondary": "MuiFormLabel-colorSecondary",
                            "disabled": "Mui-disabled Mui-disabled",
                            "error": "Mui-error Mui-error",
                            "filled": "MuiFormLabel-filled",
                            "focused": "Mui-focused Mui-focused",
                            "required": "Mui-required Mui-required",
                            "root": "MuiFormLabel-root",
                          }
                        }
                        data-shrink={true}
                        htmlFor="policyVersion"
                        id="policyVersion-label"
                      >
                        <label
                          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                          data-shrink={true}
                          htmlFor="policyVersion"
                          id="policyVersion-label"
                        >
                          Required
                          <span
                            aria-hidden={true}
                            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                      </ForwardRef(FormLabel)>
                    </WithStyles(ForwardRef(FormLabel))>
                  </ForwardRef(InputLabel)>
                </WithStyles(ForwardRef(InputLabel))>
                <WithStyles(ForwardRef(OutlinedInput))
                  autoFocus={false}
                  defaultValue="1.0.0"
                  fullWidth={false}
                  id="policyVersion"
                  label={
                    <React.Fragment>
                      Required
                       *
                    </React.Fragment>
                  }
                  multiline={false}
                  onChange={[Function]}
                >
                  <ForwardRef(OutlinedInput)
                    autoFocus={false}
                    classes={
                      Object {
                        "adornedEnd": "MuiOutlinedInput-adornedEnd",
                        "adornedStart": "MuiOutlinedInput-adornedStart",
                        "colorSecondary": "MuiOutlinedInput-colorSecondary",
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "focused": "Mui-focused",
                        "input": "MuiOutlinedInput-input",
                        "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                        "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                        "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                        "inputMultiline": "MuiOutlinedInput-inputMultiline",
                        "marginDense": "MuiOutlinedInput-marginDense",
                        "multiline": "MuiOutlinedInput-multiline",
                        "notchedOutline": "MuiOutlinedInput-notchedOutline",
                        "root": "MuiOutlinedInput-root",
                      }
                    }
                    defaultValue="1.0.0"
                    fullWidth={false}
                    id="policyVersion"
                    label={
                      <React.Fragment>
                        Required
                         *
                      </React.Fragment>
                    }
                    multiline={false}
                    onChange={[Function]}
                  >
                    <WithStyles(ForwardRef(InputBase))
                      autoFocus={false}
                      classes={
                        Object {
                          "adornedEnd": "MuiOutlinedInput-adornedEnd",
                          "adornedStart": "MuiOutlinedInput-adornedStart",
                          "colorSecondary": "MuiOutlinedInput-colorSecondary",
                          "disabled": "Mui-disabled",
                          "error": "Mui-error",
                          "focused": "Mui-focused",
                          "input": "MuiOutlinedInput-input",
                          "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                          "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                          "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                          "inputMultiline": "MuiOutlinedInput-inputMultiline",
                          "marginDense": "MuiOutlinedInput-marginDense",
                          "multiline": "MuiOutlinedInput-multiline",
                          "notchedOutline": null,
                          "root": "MuiOutlinedInput-root",
                        }
                      }
                      defaultValue="1.0.0"
                      fullWidth={false}
                      id="policyVersion"
                      inputComponent="input"
                      multiline={false}
                      onChange={[Function]}
                      renderSuffix={[Function]}
                      type="text"
                    >
                      <ForwardRef(InputBase)
                        autoFocus={false}
                        classes={
                          Object {
                            "adornedEnd": "MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd",
                            "adornedStart": "MuiInputBase-adornedStart MuiOutlinedInput-adornedStart",
                            "colorSecondary": "MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary",
                            "disabled": "Mui-disabled Mui-disabled",
                            "error": "Mui-error Mui-error",
                            "focused": "Mui-focused Mui-focused",
                            "formControl": "MuiInputBase-formControl",
                            "fullWidth": "MuiInputBase-fullWidth",
                            "input": "MuiInputBase-input MuiOutlinedInput-input",
                            "inputAdornedEnd": "MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd",
                            "inputAdornedStart": "MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart",
                            "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                            "inputMarginDense": "MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense",
                            "inputMultiline": "MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline",
                            "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                            "marginDense": "MuiInputBase-marginDense MuiOutlinedInput-marginDense",
                            "multiline": "MuiInputBase-multiline MuiOutlinedInput-multiline",
                            "root": "MuiInputBase-root MuiOutlinedInput-root",
                          }
                        }
                        defaultValue="1.0.0"
                        fullWidth={false}
                        id="policyVersion"
                        inputComponent="input"
                        multiline={false}
                        onChange={[Function]}
                        renderSuffix={[Function]}
                        type="text"
                      >
                        <div
                          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            autoFocus={false}
                            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                            defaultValue="1.0.0"
                            disabled={false}
                            id="policyVersion"
                            onAnimationStart={[Function]}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={true}
                            type="text"
                          />
                          <WithStyles(ForwardRef(NotchedOutline))
                            className="MuiOutlinedInput-notchedOutline"
                            label={
                              <React.Fragment>
                                Required
                                 *
                              </React.Fragment>
                            }
                            labelWidth={0}
                            notched={true}
                          >
                            <ForwardRef(NotchedOutline)
                              className="MuiOutlinedInput-notchedOutline"
                              classes={
                                Object {
                                  "legend": "PrivateNotchedOutline-legend-2",
                                  "legendLabelled": "PrivateNotchedOutline-legendLabelled-3",
                                  "legendNotched": "PrivateNotchedOutline-legendNotched-4",
                                  "root": "PrivateNotchedOutline-root-1",
                                }
                              }
                              label={
                                <React.Fragment>
                                  Required
                                   *
                                </React.Fragment>
                              }
                              labelWidth={0}
                              notched={true}
                            >
                              <fieldset
                                aria-hidden={true}
                                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                              >
                                <legend
                                  className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                                >
                                  <span>
                                    Required
                                     *
                                  </span>
                                </legend>
                              </fieldset>
                            </ForwardRef(NotchedOutline)>
                          </WithStyles(ForwardRef(NotchedOutline))>
                        </div>
                      </ForwardRef(InputBase)>
                    </WithStyles(ForwardRef(InputBase))>
                  </ForwardRef(OutlinedInput)>
                </WithStyles(ForwardRef(OutlinedInput))>
              </div>
            </ForwardRef(FormControl)>
          </WithStyles(ForwardRef(FormControl))>
        </ForwardRef(TextField)>
      </WithStyles(ForwardRef(TextField))>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        title="Create a new policy version from the defined parameters"
        variant="secondary"
      >
        <button
          className="btn btn-secondary"
          disabled={false}
          onClick={[Function]}
          title="Create a new policy version from the defined parameters"
          type="button"
        >
          Create New Version
        </button>
      </Button>
      <styled.div
        id="onap.policies.monitoring.tcagen2_1.0.0_org.onap.new_1.0.0"
        title="Policy Properties"
      >
        <div
          className="sc-gtsrHT duRXVT"
          id="onap.policies.monitoring.tcagen2_1.0.0_org.onap.new_1.0.0"
          title="Policy Properties"
        />
      </styled.div>
    </div>
  </styled.div>
</PolicyEditor>
`;
